### PR TITLE
lib/http/__init.py: make compatible with pyopenssl-19.1.0

### PR DESCRIPTION
**Pull request by austin987**
Initially reported at https://bugs.gentoo.org/720962

### CODE REVIEW

**Comment by rbott** [comment]
Hi @austin987 and thank you for your contribution go Ganeti! This project is currently in the process of transitioning from Google to [SPI](https://www.spi-inc.org/). We will remove the Google CLA once this process has finished as the Ganeti community has already decided not to rely on a CLA anymore. In the meantime, feel free to sign the CLA, ask other contributors who have signed the CLA to place this PR in your name or wait a bit* for the process to finish.


*) Unfortunately I do not have any idea if this will be 'days' or 'weeks' :slightly_smiling_face:

**Comment by rbott** [comment]
Just a quick update: Debian Bullseye now also ships pyopenssl 19.1.0 and hence breaks the current master branch of Ganeti. Testing against this PR works fine :+1:

File: lib/http/__init__.py [hunk 1]
```diff
@@ -89,6 +89,12 @@
 # send/receive quantum
 SOCK_BUF_SIZE = 32768
 
+# OpenSSL.SSL.ConnectionType was deprecated in pyopenssl-19.1.0:
+try:
+    ssl_conn_type = OpenSSL.SSL.Connection
+except AttributeError:
+    ssl_conn_type = OpenSSL.SSL.ConnectionType
```
Comment (iustin) [inline lib/http/__init__.py:96]: This is basically a constant, so I'd suggest instead using `SSL_CONN_TYPE`, not lowercase version.

Comment (iustin) [reply lib/http/__init__.py]: I've made a new commit that does this.

**Comment by iustin** [comment]
LGTM from my side, but I would prefer if I don't self-approve my own change :)

**Comment by rbott** [comment]
Oh I see, I can actually merge w/o having the cla:yes label. Nevermind, off it goes!

**Pull request merged**